test/unit/horizon_path_test.js

```diff
@@ -0,0 +1,241 @@
+describe("horizon path tests", function () {
+
+  beforeEach(function () {
+    this.axiosMock = sinon.mock(axios);
+    StellarSdk.Config.setDefault();
+    StellarSdk.Network.useTestNetwork();
+    let keypair = StellarSdk.Keypair.random();
+    let account = new StellarSdk.Account(keypair.publicKey(), "56199647068161");
+    this.transaction = new StellarSdk.TransactionBuilder(account)
+      .addOperation(StellarSdk.Operation.payment({
+        destination: "GASOCNHNNLYFNMDJYQ3XFMI7BYHIOCFW3GJEOWRPEGK2TDPGTG2E5EDW",
+        asset: StellarSdk.Asset.native(),
+        amount: "100.50"
+      }))
+      .build();
+    this.transaction.sign(keypair)
+
+  });
+
+  afterEach(function () {
+    this.axiosMock.verify();
+    this.axiosMock.restore();
+  });
+
+  describe('server url without folder path: https://acme.com:1337', function () {
+    let server = new StellarSdk.Server('https://acme.com:1337');
+
+    let callbuilder = null;
+
+
+
+    it("server.accounts()", function () {
+      callbuilder = server.accounts();
+      expect(callbuilder.url.toString()).to.be.equal("https://acme.com:1337/accounts");
+    });
+
+
+    it('server.accounts().accountId("fooAccountId")', function () {
+      callbuilder = server.accounts().accountId("fooAccountId");
+      callbuilder.checkFilter();
+      expect(callbuilder.url.toString()).to.be.equal("https://acme.com:1337/accounts/fooAccountId");
```

Comment:
Since you started using mock, could you remove `checkFilter` call and just check if axios was called with correct URL instead?